src/main/groovy/com/palantir/gradle/docker/DockerExtension.groovy

```diff
@@ -85,12 +87,25 @@ class DockerExtension {
         copySpec.from(files)
     }
 
-    public Set<String> getTags() {
-        return tags
+    public Set<String> getUnresolvedTags() {
+        return unresolvedTags
     }
 
+    @Deprecated
     public void tags(String... args) {
-        this.tags = ImmutableSet.copyOf(args)
+        this.unresolvedTags.addAll(args)
+    }
+
+    public Map<String, String> getTags() {
+        return ImmutableMap.copyOf(tags)
+    }
+
+    public void tag(String taskName, String tag) {
+        if (tags.containsKey(taskName)) {
```

Comment:
Consider using `putIfAbsent` instead

Reply:
Got it, I am a groovy noob, so sometimes I am using groovy like java